src/components/layout.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { CountriesContext } from '../contexts/countries';
+import { CountriesContextType, TypeCountry } from '../@types/country';
+import { useSorting } from '../hooks';
+import Button from './button';
+import Select from './select';
+import Countries from './countries';
+import IsLoading from './is-loading';
+import Error from './error';
+
+const Layout:React.FC = () =>{
+    const { countries, error, isLoading, getData } = React.useContext(CountriesContext) as CountriesContextType;
+    const [selectedContinent, setSelectedContinent] = React.useState<string | null>(null);
+    const [mobile, setMobile] = React.useState<boolean>(false);
+    
+    const query = window.matchMedia("(max-width:440px)");
+
+    const continents:string[] = Array.from(
+        new Set<string>(countries.map((country: TypeCountry) => country.continent))
+    );
+    useSorting(continents);
+    
+    const handleChange = (continent:string) => {
+        setMobile(query.matches);
+        setSelectedContinent(continent);
+    }
+
+    React.useEffect(()=>{
+        const handleChange = (event: MediaQueryListEvent) => {
+            setMobile(event.matches)
+        }     
+        query.addEventListener('change', handleChange);
+        return () => {
+           query.removeEventListener('change', handleChange)   
+        }  
+    }, [mobile])
+
+    return (
+        <div className={`container ${countries.length===0 && 'centered'}`} id={`${mobile && 'centered'}`}>
+            {!isLoading && !error && countries.length===0 && <Button text="Get Data" onClick={getData}/> }
+            {isLoading && <IsLoading />}
+            {error && <Error error={error}/>} 
```

Comment:
file is called layout which indicates it's a layout component not one that would know about countries or continents. Also mixing up countries and mobile in the id/class might be a bit messy.